crates/bdk/src/types.rs

```diff
@@ -167,7 +167,7 @@ pub struct LocalUtxo {
     /// Transaction output
     pub txout: TxOut,
     /// Type of keychain
-    pub keychain: KeychainKind,
+    //pub keychain: K,
```

Comment:
I see that, instead of returning `LocalUtxo` for the methods, you are returning a tuple `(LocalUtxo, K)` because you have removed the `keychain` field in `LocalUtxo`.

I think it would be better to keep the `keychain: K` field in `LocalUTXO<K>`, and constrain methods that can only use `K` as `KeychainKind` to return/use `LocalUtxo<KeychainKind>`.

Reply:
> Isn't this important information for an UTXO?

IMO no. These types (also LocalUTXO and WeightedUTXO) are conceptually equivalent for every keychain. They should not be generic on K. Making such primitives types generic would be awkward because we will end up having all other derived types generic too. Like `CoinSelection<K>`. It doesn't make semantic sense.

I think it's preferable to stop the generic keychain progression at `TxBuilder` level only. There are always ways to pass in the keychain with a given utxo. 

Previously it was done because it was convenient. but wasn't "necessary", and it was okay to do because keychains weren't generic.

Reply:
Resolving this as its clear from the changesets that this field is redundant.